packages/dvilela/skills/memeooorr_abci/behaviour_classes/chain.py

```diff
@@ -534,9 +534,26 @@ def async_act(self) -> Generator:
         with self.context.benchmark_tool.measure(self.behaviour_id).local():
             event = yield from self.get_event()
 
+            if event == Event.NO_FUNDS.value:
+                check_funds_count = cast(
+                    int, self.synchronized_data.check_funds_count + 1
+                )
+                self.context.logger.info(
+                    f"check_funds_count: {check_funds_count} , Event: {event}"
+                )
+                # we want to make the agent wait for 120 seconds before checking again
+                self.context.logger.info(
+                    f"Waiting for {WAIT_TIME_DELAY} seconds before checking again"
+                )
+
+                yield from self.sleep(WAIT_TIME_DELAY)
+            else:
+                check_funds_count = 0
+
```

Comment:
Even better: just define check_funds_count = 0 before the condition and remove the else. THis way, even if someone adds more code in the future, we wont find weird corner cases.